sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java

```diff
@@ -320,7 +331,7 @@ private void initializeInternal() throws IOException, UnsupportedOperationExcept
 
   private void checkEndOfRowGroup() throws IOException {
     if (rowsReturned != totalCountLoadedSoFar) return;
-    PageReadStore pages = reader.readNextFilteredRowGroup();
+    PageReadStore pages = reader.readNextRowGroup();
```

Comment:
Could you elaborate? Does `readNextRowGroup()` handle filtered row groups and column indexes, i.e. would it call `readNextFilteredRowGroup()` if column index is enabled?

Reply:
yes it will, since `ParquetRowGroupReaderImpl` calls `readNextFilteredRowGroup`. As an interface I want to make the name more general. For instance in test we can hook it up with another class that doesn't involve column index.